src/core/stdlib.js

```diff
@@ -260,6 +260,12 @@ foam.LIB({
       if ( ! a || ! b ) return false;
       if ( a.equals ) return a.equals(b);
       return a == b;
+    },
+    function compare(a, b) {
+      if ( a.compareTo ) return a.compareTo(b);
```

Comment:
Yes, add a "number" check.  Also, check if Strings are slower as well, I think they might be, and then check for them if needed also.